dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -304,7 +306,21 @@ def diff(self, other_profile: CategoricalColumn, options: dict = None) -> dict:
                     other_profile._categories.items(), key=itemgetter(1), reverse=True
                 )
             )
-
+            total_psi = 0.0
+            if cat_count1.keys() == cat_count2.keys():
+                for key in cat_count1.keys():
+                    perc_A = cat_count1[key] / self.sample_size
+                    perc_B = cat_count2[key] / other_profile.sample_size
+                    total_psi += (perc_B - perc_A) * math.log(perc_B / perc_A)
+            else:
+                warnings.warn(
```

Comment:
Maybe if we hit this we should just either 1) have PSI to None or 2) not include in the difference ducts

Reply:
I agree with this if PSI cant be calculated it shouldnt default to zero